Dataviews: Update item actions in grid view

ntsekouras opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This PR updates the item actions in DataViews to have a different design in `grid` view. Namely all the actions are inside a dropdown menu, separated by whether they are primary or not.

Additionally I have used the new Dropdown component that we use in DataViews, for consistency. I know this is going to be replaced by the ariakit version one, but when I tried adding that, I encountered way more problems right now. When the new one is stabilized, we're going to replace all usages.



## Testing Instructions
1. Enable the admin views experiment and go to manage all pages view
2. Play around with the actions in both views


## Screenshots or screencast <!-- if applicable -->
<img width="1022" alt="Screenshot 2023-11-24 at 1 28 08 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/63ebe384-3b02-4ddd-9f96-2de0f6358720">

<img width="1029" alt="Screenshot 2023-11-24 at 1 28 21 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/ecbbd06d-68ef-4df6-8a4c-8f63ddc3a3bc">

ntsekouras committed:
Dataviews: Update item actions in grid view

ciampo commented on packages/edit-site/src/components/dataviews/item-actions.js:
```diff
@@ -2,18 +2,28 @@
+import { unlock } from '../../lock-unlock';
+
+const {
+	DropdownMenuV2: DropdownMenu,
```
Would it be possible to use `DropdownMenuV2Ariakit` & related subcomponents instead? 

`DropdownMenuV2` (the radix-based version) is not maintained anymore. I'm in the process of migrating all existing usages to the ariakit-based version, and I'll remove this component completely once the migration is over. (components naming will also improve after that!)

There are minor differences in terms of APIs, you can take a look at the [Storybook file](https://github.com/WordPress/gutenberg/blob/fc5619d6ac11605683f4a72dcea53e92d20f4410/packages/components/src/dropdown-menu-v2-ariakit/stories/index.story.tsx) to see a few examples.

ntsekouras replied:
I'll give it a go again. I tried at first and had some issues, that I'll remember if I try again 😆 . I think it had to do with `align` and placement and something with the trigger I think. I'll post soon with the exact issues.

ciampo replied:
In the new component, you should use the `placement` prop (same as our `Popover`) instead of the `align` prop, but let me know in case there's more. Also happy to pair on it quickly if that helps.

Getting feedback from folks like you when using the component is actually very useful :)

ntsekouras replied:
@ciampo I pushed an update to use the new component, but I can't make it look like this(alignment)
<img width="344" alt="Screenshot 2023-11-24 at 3 47 33 PM" src="https://github.com/WordPress/gutenberg/assets/16275880/5351236d-b38e-48eb-9d08-635a0a02ce30">

Also regarding my initial testing, while the `placement` seems to work well, in story book it doesn't. I had checked that without trying it code and missed that it's a problem in story book.

ciampo replied:
> while the placement seems to work well, in story book it doesn't

Could you be more specific?

Also, there may be two factors that affect testing `placement` in Storybook:
- a `floating-ui` bug which causes all popover-based components to have an extra offset when the left sidebar is opened. This bug can be avoided by closing the left sidebar when testing the Storybook examples.
- The fact that the dropdown trigger (button) is rendered in the top left of the viewport means that a lot of placements are going to naturally "flip" because there isn't enough space. I suggest adding some margin/padding around the button to test such placements

Let me know if you're still facing issues, though!

ntsekouras committed:
Use new dropdown

jameskoster commented:
Seems to be working well for me:

<img width="200" alt="Screenshot 2023-11-24 at 14 27 10" src="https://github.com/WordPress/gutenberg/assets/846565/8416722b-b365-4a14-9c26-66cf25d52674">

Seeing the menu in practise makes me think we can probably simplify by removing the icons and separator. Those are likely to be more useful when additional secondary actions are added.

<img width="199" alt="Screenshot 2023-11-24 at 14 26 37" src="https://github.com/WordPress/gutenberg/assets/846565/c336920b-8d85-4d1c-82e2-448c3d16b488">

Side note: The ellipsis icon in the trigger seems to be blue in its resting state. Generally icon buttons are dark gray (`$gray-900`). Unsure if that's an issue here or in the new component.

ntsekouras commented:
>Seems to be working well for me:

Hm.. in Chrome for me it isn't, but tested in FF and is displayed as I expected.

--edit
I updated Chrome and it works properly now. Not sure if there was some local issue, or it's not working well with a bit older Chrome version.

ciampo commented:
It's likely that you were previously running  Chrome 116 (or earlier), since support for CSS Subgrid was added in Chrome 117.

ntsekouras committed:
remove separators and icons

ntsekouras commented:
I removed the icons and the separators.

>Side note: The ellipsis icon in the trigger seems to be blue in its resting state. Generally icon buttons are dark gray ($gray-900). Unsure if that's an issue here or in the new component.

That was because of `variant="tertiary"`, which I removed. When is expanded though the color comes from the Button(`aria-expanded=true`), which is probably good, no? @ciampo

youknowriad commented on packages/edit-site/src/components/dataviews/view-grid.js:
```diff
@@ -61,6 +61,7 @@ export function ViewGrid( { data, fields, view, actions, getItemId } ) {
 								<ItemActions
 									item={ item }
 									actions={ actions }
+									viewType={ view.type }
```
Minor: Maybe instead of passing view.type, we pass a flag that says `compact`  or not. That way any view type can just choose by passing true or false to this prop instead of having to hard coded the view types within the shared component.

ntsekouras replied:
That's a good thought, yes. I'll update.

youknowriad commented:
Am I not seeing right or the border radius of that dropdown is different than the "view" dropdown for instance? Should we unify everything?

youknowriad commented:
Also, not the same hover styles... Why two different dropdown styles?

ntsekouras commented:
>Also, not the same hover styles... Why two different dropdown styles?

In this PR I'm using the new Dropdown menu and @ciampo has a PR to update all the other usages in a [different PR.](https://github.com/WordPress/gutenberg/pull/55625#issuecomment-1822355621)

youknowriad reviewed (approved):
The style change is weird but I'm proving as we're bringing consistency later.

ntsekouras committed:
add `isCompact` prop instead of viewType

ciampo commented:
> I removed the icons and the separators.
> 
> > Side note: The ellipsis icon in the trigger seems to be blue in its resting state. Generally icon buttons are dark gray ($gray-900). Unsure if that's an issue here or in the new component.
> 
> That was because of `variant="tertiary"`, which I removed. When is expanded though the color comes from the Button(`aria-expanded=true`), which is probably good, no? @ciampo

The new `DropdownMenu` component shouldn't apply any styles in particular to the trigger — therefore, those styles should be all coming from the `Button` component. But if you ever notice something different, please let me know
